test/e2e/tests/confirmations/header.spec.js

```diff
@@ -0,0 +1,88 @@
+const { strict: assert } = require('assert');
+const {
+  defaultGanacheOptions,
+  openDapp,
+  unlockWallet,
+  WINDOW_TITLES,
+  withFixtures,
+} = require('../../helpers');
+const FixtureBuilder = require('../../fixture-builder');
+
+describe('Header component', function () {
+  if (!process.env.ENABLE_CONFIRMATION_REDESIGN) {
+    return;
+  }
+
+  const ONE_CLICK_CONFIRMATIONS_USING_BLOCKAID = [
+    { name: 'Personal Sign signature', testDAppBtnId: 'personalSign' },
+    { name: 'Sign Typed Data signature', testDAppBtnId: 'signTypedData' },
+    { name: 'Sign Typed Data v3 signature', testDAppBtnId: 'signTypedDataV3' },
+    { name: 'Sign Typed Data v4 signature', testDAppBtnId: 'signTypedDataV4' },
+    { name: 'Sign Permit signature', testDAppBtnId: 'signPermit' },
+  ];
+
+  ONE_CLICK_CONFIRMATIONS_USING_BLOCKAID.forEach((confirmation) => {
+    it(`${confirmation.name} component includes header with balance and copyable address element`, async function () {
+      await withFixtures(
+        {
+          dapp: true,
+          fixtures: new FixtureBuilder()
+            .withPermissionControllerConnectedToTestDapp()
+            .withPreferencesController({
+              preferences: { redesignedConfirmations: true },
+            })
+            .build(),
+          ganacheOptions: defaultGanacheOptions,
+          title: this.test?.fullTitle(),
+        },
+        async ({ driver }) => {
+          await unlockWallet(driver);
+          await openDapp(driver);
+
+          const WALLET_ADDRESS = '0x5CfE73b6021E818B776b421B1c4Db2474086a7e1';
+
+          await clickConfirmationBtnOnTestDapp(
+            driver,
+            confirmation.testDAppBtnId,
+          );
+          await clickHeaderInfoBtnAndAssertBalance(driver);
+          await copyAddressAndAssertPastedValue(driver, WALLET_ADDRESS);
+        },
+      );
+    });
+
+    async function clickConfirmationBtnOnTestDapp(driver, btnId) {
+      await driver.switchToWindowWithTitle(WINDOW_TITLES.TestDApp);
+      await driver.clickElement(`#${btnId}`);
+      await driver.delay(2000);
```

Comment:
Adding fixed delays like this will cause tests to be flaky, can we make this wait a bit smarter? like waiting for an element or something?

Reply:
@pnarayanaswamy thanks for reviewing! This delay is precisely intended to prevent test flakiness that sometimes happens on opening of the confirmation window

```js
await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
```

Unfortunately, I haven't found a more reliable pattern for doing this.